File: Premtek.Base/CLaserReaderCards.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Xml.Linq;
using System.Threading.Tasks;
using ProjectCore;
using Premtek;

namespace Premtek.Base
{

	/// <summary>雷射干涉儀控制器連線設定</summary>
	/// <remarks></remarks>
	public class CLaserReaderCards
	{
		public int Count {
			get { return Parameters.Count; }
		}
		public List<sLaserReaderConnectionParameter> Parameters = new List<sLaserReaderConnectionParameter>();
		public sLaserReaderConnectionParameter this[int index] {
			get { return Parameters[index]; }
			set { Parameters[index] = value; }
		}

		/// <summary>儲存連線參數設定</summary>
		/// <param name="fileName"></param>
		/// <returns></returns>
		/// <remarks></remarks>
		public object Save(string fileName)
		{
			int CardCount = Parameters.Count;
			string mSection = "LaserReader";
			CIni.SaveIniString(mSection, "CardCount", CardCount, fileName);
			//卡片數量儲存

			for (int mCardNo = 0; mCardNo <= CardCount - 1; mCardNo++) {
				mSection = "Laser" + (mCardNo + 1).ToString() + "-Connection";
                CIni.SaveIniString(mSection, "CardType", Convert.ToInt32(Parameters[mCardNo].CardType), fileName);

                CIni.SaveIniString(mSection, "DL-RS1A-PortName", Parameters[mCardNo].DLRS1A.PortName, fileName);
                CIni.SaveIniString(mSection, "DL-RS1A-BaudRate", Parameters[mCardNo].DLRS1A.BaudRate, fileName);

                CIni.SaveIniString(mSection, "IPAddress", Parameters[mCardNo].LJV7060TCP.IP, fileName);
                CIni.SaveIniString(mSection, "Port", Parameters[mCardNo].LJV7060TCP.Port, fileName);

			}

			return true;
		}
		/// <summary>讀取連線參數設定</summary>
		/// <param name="fileName"></param>
		/// <returns></returns>
		/// <remarks></remarks>
		public object Load(string fileName)
		{
			string mSection = "LaserReader";
			int mCardCount = 0;
			mCardCount = Convert.ToInt32(CIni.ReadIniString(mSection, "CardCount", fileName,((int)MCommonDefine.enmLaserReader.Max + 1).ToString()));
			Parameters.Clear();

			for (int mLaserNo = 0; mLaserNo <= mCardCount - 1; mLaserNo++) {
				mSection = "Laser" + (mLaserNo + 1).ToString() + "-Connection";
				sLaserReaderConnectionParameter mItem = new sLaserReaderConnectionParameter();

                mItem.CardType = (MSystemParameter.enmLaserInterferometerType)Convert.ToInt16(CIni.ReadIniString(mSection, "CardType", fileName, 0));

                mItem.DLRS1A.PortName = CIni.ReadIniString(mSection, "DL-RS1A-PortName", fileName, "COM1");
                mItem.DLRS1A.BaudRate = CIni.ReadIniString(mSection, "DL-RS1A-BaudRate", fileName, "9600");

                mItem.LJV7060TCP.IP = CIni.ReadIniString(mSection, "IPAddress", fileName, "");
                mItem.LJV7060TCP.Port = Convert.ToInt16(CIni.ReadIniString(mSection, "Port", fileName, 0));

				Parameters.Add(mItem);
			}
			return true;
		}

	}
}

File: ProjectLaserInterferometer/CLaserReaderCards.vb
Imports ProjectCore

''' <summary>雷射干涉儀控制器連線設定</summary>
''' <remarks></remarks>
Public Class CLaserReaderCards
    Public ReadOnly Property Count As Integer
        Get
            Return Parameters.Count
        End Get
    End Property
    Public Parameters As New List(Of sLaserReaderConnectionParameter)
    Default Public Property Items(ByVal index As Integer) As sLaserReaderConnectionParameter
        Get
            Return Parameters(index)
        End Get
        Set(value As sLaserReaderConnectionParameter)
            Parameters(index) = value
        End Set
    End Property

    ''' <summary>儲存連線參數設定</summary>
    ''' <param name="fileName"></param>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Function Save(ByVal fileName As String)
        Dim CardCount As Integer = Parameters.Count
        Dim mSection As String = "LaserReader"
        SaveIniString(mSection, "CardCount", CardCount, fileName) '卡片數量儲存

        For mCardNo As Integer = 0 To CardCount - 1
            mSection = "Laser" & (mCardNo + 1).ToString & "-Connection"
            With Parameters(mCardNo)
                SaveIniString(mSection, "CardType", CInt(.CardType), fileName)

                SaveIniString(mSection, "DL-RS1A-PortName", .DLRS1A.PortName, fileName)
                SaveIniString(mSection, "DL-RS1A-BaudRate", .DLRS1A.BaudRate, fileName)

                SaveIniString(mSection, "IPAddress", .LJV7060TCP.IP, fileName)
                SaveIniString(mSection, "Port", .LJV7060TCP.Port, fileName)

            End With
        Next

        Return True
    End Function
    ''' <summary>讀取連線參數設定</summary>
    ''' <param name="fileName"></param>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Function Load(ByVal fileName As String)
        Dim mSection As String = "LaserReader"
        Dim mCardCount As Integer
        mCardCount = Val(ReadIniString(mSection, "CardCount", fileName, enmLaserReader.Max + 1))
        Parameters.Clear()

        For mLaserNo As Integer = 0 To mCardCount - 1
            mSection = "Laser" & (mLaserNo + 1).ToString & "-Connection"
            Dim mItem As New sLaserReaderConnectionParameter
            With mItem
                .CardType = Val(ReadIniString(mSection, "CardType", fileName, 0))

                .DLRS1A.PortName = ReadIniString(mSection, "DL-RS1A-PortName", fileName, "COM1")
                .DLRS1A.BaudRate = ReadIniString(mSection, "DL-RS1A-BaudRate", fileName, "9600")

                .LJV7060TCP.IP = ReadIniString(mSection, "IPAddress", fileName)
                .LJV7060TCP.Port = Val(ReadIniString(mSection, "Port", fileName))

            End With
            Parameters.Add(mItem)
        Next
        Return True
    End Function

End Class
